crates/api_crud/src/community/delete.rs

```diff
@@ -22,13 +22,13 @@ pub async fn delete_community(
   local_user_view: LocalUserView,
 ) -> LemmyResult<Json<CommunityResponse>> {
   // Fetch the community mods
-  let community_id = data.community_id;
   let community_mods =
-    CommunityModeratorView::for_community(&mut context.pool(), community_id).await?;
+    CommunityModeratorView::for_community(&mut context.pool(), data.community_id).await?;
 
+  let community = Community::read(&mut context.pool(), data.community_id).await?;
```

Comment:
No need to read this, you could get it from the first result of the above query, because `for_community` already returns communities.  

The only possible bug there is if there's a modless community for some reason, which we'd need to check for in the mod adding / removing actions, not here.

Reply:
That would need unwrap or similar because for Rust an array can always be empty. Not clean at all imo.